doc/source/dev_guide/plugins.rst

```diff
@@ -1,34 +1,207 @@
+
+   Lastly, this directory should be structured like a standard python package.
+   This primarily means a ``mypkg/__init__.py`` file should exist.
+
+pyproject.toml
+--------------
+
+A ``pyproject.toml`` file can be used to define the metadata and configuration
+for a python package. With this file it is possible to use package building
+tools to make an installable package. By using a special feature called
+"entry points" we can configure our package to its satpy features are
+automatically discovered by Satpy.
+
+A ``pyproject.toml`` file is typically placed in the root of a project
+repository and at the same level as the package (ex. ``satpy_myplugin/``
+directory). An example for a package called ``satpy-myplugin`` with
+custom composites is shown below.
+
+.. code:: toml
+
+    [project]
+    name = "satpy-myplugin"
+    description = "Example Satpy plugin package definition."
+    version = "1.0.0"
+    readme = "README.md"
+    license = {text = "GPL-3.0-or-later"}
+    requires-python = ">=3.8"
+    dependencies = [
+        "satpy",
+    ]
+
+    [tool.setuptools]
+    packages = ["satpy_myplugin"]
+
+    [build-system]
+    requires = ["setuptools", "wheel"]
+    build-backend = "setuptools.build_meta"
+
+    [project.entry-points."satpy.composites"]
+    example_composites = "satpy_myplugin"
+
+Other custom components like readers and writers can be defined in the same
+package by using additional entry points named ``satpy.readers`` for readers,
+``satpy.writers`` for writers, and ``satpy.enhancements`` for enhancements.
+
+Note the difference between the usage of the package name (``satpy-myplugin``)
+which includes a hyphen and the package directory (``satpy_myplugin``) which uses
+an underscore. Your package name does not need to have a separator (hyphen) in
+it, but is used here due to the common practice of naming plugins this way.
+Package directories can't use hyphens as this would be a syntax error when
+trying to import the package. Underscores can't be used in package names as
+this is not allowed by PyPI.
+
+The first ``project`` section in this TOML file specifies metadata about the
+package. This is most important if you plan on distributing your package on
+PyPI or similar package repository. We specify that our package depends on
+``satpy`` so if someone installs it Satpy will automatically be installed.
+The second ``tools.setuptools`` section
+tells the package building (via ``setuptools``) what directory the Python
+code is in. The third section, ``build-system``, says what tool(s) should be
+used for building the package and what extra requirements are needed during
+this build process.
+
+The last section, ``project.entry-points."satpy.composites"`` is the only
+section specific to this package being a Satpy plugin. At the time of writing
+the ``example_composites = "satpy_myplugin"`` portion is not actually used
+by Satpy. Instead Satpy will assume that a package that defines the
```

Comment:
`Instead Satpy will assume ...` this creates a causality between the fact that the entry point metadata is not used and the fact that the yaml file directories are to be structured as in satpy. I know that you wanted this metadata to provide the yaml directories, but I think this causality should be removed as it might not make sense to users reading this.

Reply:
> I know that you wanted this metadata to provide the yaml directories

You mean how I wanted the entry point (`example_composites = "satpy_myplugin"`) to point to a package-relative directory for the YAML? Why would this be confusing? How does that change what is said in this paragraph?

Reply:
I'm happy with the new formulation.